facebook/OpenBIC
# Artemis: cb: Support to set power monitor alert threshold

# Description
- Support to set INA233 alert threshold.
- Support to enable SQ52205 alert event and set alert threshold.

# Motivation
- Support to set INA233 alert threshold.
- Support to enable SQ52205 alert event and set alert threshold.

# Test Plan
- Build code: Pass in AT-CB/AT-MC/OP-OP2/YV35-RF/YV35-NF
- Check INA233 alert threshold on main source ACB: Pass
- Check SQ52205 enable alert event on main source ACB: Pass
- Check SQ52205 alert threshold on main source ACB: Pass

# Log
- Check INA233 alert threshold (ACCL1 and ACCL12 not present) uart:~$ i2c scan I2C_3
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:             -- -- -- -- -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: 40 41 42 43 44 45 -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --
6 devices found on I2C_3
uart:~$ i2c read I2C_3 0x41 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_3 0x42 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_3 0x43 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_3 0x44 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_3 0x45 0x6B 0x02 // Power-over threshold
00000000: 00 2b

uart:~$ i2c scan I2C_8
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:             -- -- -- -- -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: 40 41 42 43 44 45 -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --
6 devices found on I2C_8
uart:~$ i2c read I2C_8 0x40 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_8 0x41 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_8 0x42 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_8 0x43 0x6B 0x02 // Power-over threshold
00000000: 00 2b
uart:~$ i2c read I2C_8 0x44 0x6B 0x02 // Power-over threshold
00000000: 00 2b

- Check SQ52205 enable alert event and alert threshold uart:~$ i2c scan I2C_1
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:             -- -- -- -- -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: 40 41 -- -- -- -- -- -- 48 -- 4a -- -- -- -- --
50: -- -- -- -- -- -- 56 -- -- 59 -- -- -- -- 5e --
60: -- -- -- -- -- -- -- -- -- -- 6a -- 6c -- -- 6f
70: 70 -- -- -- -- -- -- --
11 devices found on I2C_1
uart:~$ i2c read I2C_1 0x40 0x06 0x02 // Enable Power-over event
00000000: 08 48
uart:~$ i2c read I2C_1 0x40 0x07 0x02 // Alert threshold
00000000: 05 1d
uart:~$ i2c read I2C_1 0x41 0x06 0x02 // Enable Power-over event
00000000: 08 08
uart:~$ i2c read I2C_1 0x41 0x07 0x02 // Alert threshold
00000000: 05 1d
